newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudResourceMover.js

```diff
@@ -45,59 +52,105 @@ export const moveAllCloudProjectResourcesToCloudProject = async ({
    * Find the resources stored on GDevelop Cloud that must be downloaded and
    * uploaded into the new project.
    */
-  const getResourcesToFetchAndUpload = (
+  const getResourcesToFetchAndUpload = async (
     project: gdProject
-  ): Array<ResourceToFetchAndUpload> => {
+  ): Promise<Array<ResourceToFetchAndUpload>> => {
     const resourcesManager = project.getResourcesManager();
     const allResourceNames = resourcesManager.getAllResourceNames().toJSArray();
-    return allResourceNames
-      .map(
-        (resourceName: string): ?ResourceToFetchAndUpload => {
-          const resource = resourcesManager.getResource(resourceName);
-          const resourceFile = resource.getFile();
-
-          if (isURL(resourceFile)) {
-            if (checkIfIsGDevelopCloudBucketUrl(resourceFile)) {
-              if (
-                extractProjectUuidFromProjectResourceUrl(resourceFile) ===
-                newCloudProjectId
-              ) {
-                // Somehow the resource is *already* stored in the new project - surely because
-                // the project resources were partially moved (like when you click "Retry" after some failures
-                // when saving a project as a new cloud project).
-                // Just ignore this resource which is already moved then.
-                return null;
-              }
-
-              return {
-                resource,
-                url: resourceFile,
-                filename: extractFilenameFromProjectResourceUrl(resourceFile),
-              };
-            } else if (isBlobURL(resourceFile)) {
-              result.erroredResources.push({
-                resourceName: resource.getName(),
-                error: new Error('Unsupported blob URL.'),
-              });
-              return null;
-            } else {
-              // Public URL resource: nothing to do.
-              return null;
+    const resourceToFetchAndUpload: Array<ResourceToFetchAndUpload> = [];
+
+    let tokenForPrivateGameTemplateAuthorization = null;
+    let isFetchingGameTemplateAuthorizedResources = false;
+    for (const resourceName of allResourceNames) {
+      const resource = resourcesManager.getResource(resourceName);
+      const resourcePath = resource.getFile();
+      if (
+        isURL(resourcePath) &&
+        isPrivateGameTemplateResourceAuthorizedUrl(resourcePath)
+      ) {
+        isFetchingGameTemplateAuthorizedResources = true;
+      }
+    }
+    if (isFetchingGameTemplateAuthorizedResources) {
+      const userId = authenticatedUser.profile && authenticatedUser.profile.id;
+      if (!userId) {
+        throw new Error(
+          'Can not fetch resources from a private game template without being authenticated.'
+        );
+      }
+      tokenForPrivateGameTemplateAuthorization = await getAuthorizationTokenForPrivateGameTemplates(
+        authenticatedUser.getAuthorizationHeader,
+        { userId }
+      );
+    }
```

Comment:
See if you can move this into a helper function, so the reader can just read the function name and understand this is "just" getting the tokenFroPrivateGameTemplateAuthorization, if needed (and if not that's fine)